tests/test_notifications.py

```diff
@@ -796,6 +725,242 @@ def test_notifications_dict_add_message_returns_proper_format(self):
         assert_equal(d, expected)
 
 
+class TestCompileSubscriptions(OsfTestCase):
+    def setUp(self):
+        super(TestCompileSubscriptions, self).setUp()
+        self.user_1 = factories.UserFactory()
+        self.user_2 = factories.UserFactory()
+        self.user_3 = factories.UserFactory()
+        self.user_4 = factories.UserFactory()
+        # Base project + 1 project shared with 3 + 1 project shared with 2
+        self.base_project = factories.ProjectFactory(is_public=False, creator=self.user_1)
+        self.shared_node = factories.NodeFactory(parent=self.base_project, is_public=False, creator=self.user_1)
+        self.private_node = factories.NodeFactory(parent=self.base_project, is_public=False, creator=self.user_1)
+        # Adding contributors
+        for node in [self.base_project, self.shared_node, self.private_node]:
+            node.add_contributor(self.user_2, permissions='admin')
+        self.base_project.add_contributor(self.user_3, permissions='write')
+        self.shared_node.add_contributor(self.user_3, permissions='write')
+        # Setting basic subscriptions
+        self.base_sub = factories.NotificationSubscriptionFactory(
+            _id=self.base_project._id + '_file_updated',
+            owner=self.base_project,
+            event_name='file_updated'
+        )
+        self.base_sub.save()
+        self.shared_sub = factories.NotificationSubscriptionFactory(
+            _id=self.shared_node._id + '_file_updated',
+            owner=self.shared_node,
+            event_name='file_updated'
+        )
+        self.shared_sub.save()
+        self.private_sub = factories.NotificationSubscriptionFactory(
+            _id=self.private_node._id + '_file_updated',
+            owner=self.private_node,
+            event_name='file_updated'
+        )
+        self.private_sub.save()
+
+    def test_no_subscription(self):
+        node = factories.NodeFactory()
+        result = emails.compile_subscriptions(node, 'file_updated')
+        assert_equal({'email_transactional': [], 'none': [], 'email_digest': []}, result)
+
+    def test_no_subscribers(self):
+        node = factories.NodeFactory()
+        node_sub = factories.NotificationSubscriptionFactory(
+            _id=node._id + '_file_updated',
+            owner=node,
+            event_name='file_updated'
+        )
+        node_sub.save()
+        result = emails.compile_subscriptions(node, 'file_updated')
+        assert_equal({'email_transactional': [], 'none': [], 'email_digest': []}, result)
+
+    def test_creator_subbed_parent(self):
+        """Basic sub check"""
```

Comment:
As per previous comments and our style guide, prefer code comments to docstrings *in tests*.